pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -570,6 +594,42 @@ func UnqualifiedSearchRegistries(ctx *types.SystemContext) ([]string, error) {
 	return config.UnqualifiedSearchRegistries, nil
 }
 
+// UnqualifiedSearchRegistriesWithOrigin returns a list of host[:port] entries
+// to try for unqualified image search, in the returned order.  It also returns
+// a human-readable description of where these entries are specified (e.g., a
+// registries.conf file).
+func UnqualifiedSearchRegistriesWithOrigin(ctx *types.SystemContext) ([]string, string, error) {
+	config, err := getConfig(ctx)
+	if err != nil {
+		return nil, "", err
+	}
+	return config.UnqualifiedSearchRegistries, config.unqualifiedSearchRegistriesOrigin, nil
+}
+
+// parseShortNameMode translates the string into well-typed
+// types.ShortNameMode.
+func parseShortNameMode(mode string) (types.ShortNameMode, error) {
+	switch mode {
+	case "disabled":
+		return types.ShortNameModeDisabled, nil
+	case "enforcing":
+		return types.ShortNameModeEnforcing, nil
+	case "permissive", "":
+		return types.ShortNameModePermissive, nil
+	default:
+		return types.ShortNameModeInvalid, errors.Errorf("invalid short-name mode: %q", mode)
+	}
+}
+
+// GetShortNameMode returns the types.ShortNameMode in the config files.
```

Comment:
Something like
```suggestion
// GetShortNameMode returns the configured types.ShortNameMode .
```
or s/configured/applicable/ ? It returns a value even if config files don’t include a value.